packages/core/src/middleware/koa-slonik-error-handler.ts

```diff
@@ -39,12 +39,16 @@ export default function koaSlonikErrorHandler<StateT, ContextT>(): Middleware<St
         case InsertionError:
           throw new RequestError({
             code: 'entity.create_failed',
+            // Error constructor does not assert the instance type
+            // eslint-disable-next-line no-restricted-syntax
             name: (error as InsertionError<SchemaLike>).schema.tableSingular,
           });
```

Comment:
how about we just simply use `if (error instanceof Foo)` here?

Reply:
Tried, `instanceof Class` does not accept a Generic type. like `instanceof InsertionError<SchemaLike>`. Without it, `error.schema` is the type of any, have to use as to assert a Generic type anyway.

Reply:
hold on, let me try something

Reply:
<img width="587" alt="image" src="https://user-images.githubusercontent.com/14722250/179904010-728b6f68-5393-4227-9635-d1c9154ea2a7.png">

then let's use if + as, which looks more clear

Reply:
updated. Since quite straightforward. address this conversation directly.